### DISCUSSION TITLE
Get Binary Bias Values

### DISCUSSION THREAD

**Discussion by Sairam954** [larq/larq#772] [Q&A]
Hi Team,

I am trying to access binary bias values of the trained larq model, I am able to access weights with the below code in the tutorial. Can you please help me with bias values?
```python
with larq.context.quantized_scope(True):
    
    weights = model.get_weights()  # get binary weights
    print(np.shape(model.layers[1].get_weights()))
```

**Comment by CNugteren** [comment]
I believe by calling that function you get both the weights and the bias terms. I combined the tutorial network with your code and modified it slightly:
```python
import tensorflow as tf
import numpy as np
import larq

model = tf.keras.models.Sequential([
        tf.keras.layers.Flatten(),
        larq.layers.QuantDense(512, kernel_quantizer="ste_sign", kernel_constraint="weight_clip"),
        larq.layers.QuantDense(10, input_quantizer="ste_sign", kernel_quantizer="ste_sign", kernel_constraint="weight_clip", activation="softmax"),
])
model.build(input_shape=(32, 32))

with larq.context.quantized_scope(True):
    weights = model.layers[1].get_weights()
    print(weights[0])  # (32, 512) binarized weights
    print(weights[1])  # (512,) floating-point bias terms
```

**Reply by Sairam954** [reply]
Thanks, @CNugteren, how can we obtain binary bias terms? or bias is still in floating-point?

**Reply by CNugteren** [reply]
I guess you could make the bias terms binary as well, but it doesn't make much sense to me: you've done hundreds of multiply-accumulates to end up with a possibly large value and then you are going to add or subtract just 1. No I think depending on the size of your later you'll want at least 16 or 32 bits integer or a floating point value as bias.

**Reply by martinvernengo** [reply]
Hence no scaling factor is used por the binary weights? As far as I could see from the [this paper](https://arxiv.org/abs/1603.05279) by Rastegari et al., the FP bias terms are included in the batch normalization layer and the XNOR net utilizes FP scaling factors that multiply the binary {-1, 1} weights. In case the answer is YES to my question, is there any way to program larq so that it DOES use the scaling factors and the bathc normalization for the bias part?

**Reply by CNugteren** [reply]
@martinvernengo , your question is answered in https://github.com/larq/larq/discussions/908.
